qcodes/dataset/sqlite/queries.py

```diff
@@ -1086,7 +1066,7 @@ def get_last_experiment(conn: ConnectionPlus) -> int | None:
     return c.fetchall()[0][0]
 
 
-def get_runs(conn: ConnectionPlus, exp_id: int | None = None) -> list[sqlite3.Row]:
+def get_runs(conn: ConnectionPlus, exp_id: int | None = None) -> list[tuple[int]]:
```

Comment:
This function is used only here:
https://github.com/QCoDeS/Qcodes/blob/7270bca1d744d2dd460550d76a7d276af02e9895/qcodes/dataset/experiment_container.py#L170-L175
It's not an optimization, but there is no point to expose internal structure (here the SQL table) while only one column is relevant.

For that one and https://github.com/QCoDeS/Qcodes/pull/4446#discussion_r970578271 I can keep `"SELECT * FROM"` and `tuple[X,...]` if you prefer.

Reply:
ok, i am probably over-worried about changing things in a module that has been considered private but that will only become official in the upcoming qcodes release.

> there is no point to expose internal structure (here the SQL table) while only one column is relevant

i agree, so let's keep the change, but also improve the return to be a list of ints, so without the tuple wrapping (because it does not add any value).